aws/rust-runtime/aws-config/src/profile/credentials/repr.rs

```diff
@@ -209,13 +216,19 @@ mod static_credentials {
     pub const AWS_SECRET_ACCESS_KEY: &str = "aws_secret_access_key";
     pub const AWS_SESSION_TOKEN: &str = "aws_session_token";
 }
+
+mod credential_process {
+    pub const CREDENTIAL_PROCESS: &str = "credential_process";
+}
 const PROVIDER_NAME: &str = "ProfileFile";
 
 fn base_provider(profile: &Profile) -> Result<BaseProvider, ProfileFileError> {
+    dbg!("checking base provider", &profile);
     // the profile must define either a `CredentialsSource` or a concrete set of access keys
     match profile.get(role::CREDENTIAL_SOURCE) {
         Some(source) => Ok(BaseProvider::NamedSource(source)),
-        None => web_identity_token_from_profile(profile)
+        None => credential_process_from_profile(profile)
+            .or_else(|| web_identity_token_from_profile(profile))
             .or_else(|| sso_from_profile(profile))
             .unwrap_or_else(|| Ok(BaseProvider::AccessKey(static_creds_from_profile(profile)?))),
     }
```

Comment:
I'm not sure what the usual ordering is for the SDKs. I assume the Java one is pretty canonical but lacks mention of `credential_process`: https://docs.aws.amazon.com/sdk-for-java/v1/developer-guide/credentials.html

Reply:
I happen to be looking at some functionality in a similar area (I stumbled on your PR while checking for any existing efforts on MFA), and I've been cribbing from `botocore`. I'm sure someone from the team will be able to do better but I thought I might as well pass over my notes, since I was just in the area - hopefully this helps. I agree with the general sentiment that the ordering should be consistent between SDKs - so I would guess it makes sense to put `credentials_process` after SSO?

---

`botocore` looks like it has the ordering for profiles as](https://github.com/boto/botocore/blob/develop/botocore/credentials.py#L110):

1. env
2. assume role
3. [profile](https://github.com/boto/botocore/blob/develop/botocore/credentials.py#L167):
    1. web
    4. sso
    5. credentials process
    6. config
4. legacy ec2
5. boto
6. container
7. instance metadata

I don't have a local checkout of the Java SDK, but I think [the equivalent section for profiles is here](https://github.com/aws/aws-sdk-java/blob/fd409dee8ae23fb8953e0bb4dbde65536a7e0514/aws-java-sdk-core/src/main/java/com/amazonaws/auth/profile/ProfilesConfigFile.java#L211), which goes in the order:

1. assume role
2. credentials process
3. static credentials (equiv. to boto's "config" in profile handling?)

Reply:
Good find, thanks @jelford ! It seems sensible to follow `boto` here. I'll move `credential_process` down below `sso`.

Reply:
yup, lets follow boto

Reply:
yup, lets follow boto

Reply:
Reordered in [14abfdd](https://github.com/awslabs/smithy-rs/pull/1356/commits/14abfdd7811afc6aaf65765956919c9fdb1989b9)